src/version.ts

```diff
@@ -1,6 +1,7 @@
 import { Error } from "./errors";
 
-const versionRegexp = /^(\d+)\.(\d+).(\d+)(-dev)?$/;
+// regular expression suggested by semver.org
+const versionRegexp = /^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/;
 
```

Comment:
Can we just use a simple one that will cover what we need? This is completely unreadable by humans 😂. I doubt we need to support the whole sem ver spec.